microscopic-monks/primal/engine/camera.py

```diff
@@ -0,0 +1,80 @@
+from typing import Tuple
+from kivy.graphics.transformation import Matrix
+from kivy.graphics.context_instructions import MatrixInstruction, PushMatrix, PopMatrix
+
+
+class Camera:
+    def get_viewport(self) -> Tuple[int, int]:
+        raise NotImplementedError
+
+    def get_projection(self) -> Matrix:
+        raise NotImplementedError
+
+
+class OrthographicCamera(Camera):
```

Comment:
Realy well written camera class. It is easy to read, it has short methods, and also som annotations. I like in genrall how the projection works.
I would love to see some docstrings in such a well written class.

Reply:
Thanks. The lack of docstrings is big fault of this project. I would like to write them, but I was under time pressure since I was implementing a lot of stuff.

Reply:
> Thanks. The lack of docstrings is big fault of this project. I would like to write them, but I was under time pressure since I was implementing a lot of stuff.

I've seen this mentioned in other projects as well. My advise would be to write the docstrings *before* writing the code implementation; this can help significantly with improving the overall organization, and can actually reduce overall development time since it keeps things much more focused.